sky/packages/sky/lib/src/fn3/focus.dart

```diff
@@ -137,8 +140,9 @@ class FocusState extends State<Focus> {
   void _scopeRemoved(GlobalKey key) {
     assert(_focusedScope == key);
     _currentlyRegisteredScopeRemovalListenerKey = null;
-    setState(() {
-      _focusedScope = null;
+    _focusedScope = null;
```

Comment:
If we happen to already be dirty, then we'll update this frame, otherwise, we'll update next frame. That seems weird and unsatisfying.